src/main/java/com/khmall/domain/category/CategoryPathService.java

```diff
@@ -0,0 +1,121 @@
+package com.khmall.domain.category;
+
+import com.khmall.domain.category.CategoryRepository.Flat;
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Deque;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+@Transactional(readOnly = true)
+public class CategoryPathService {
+
+  private final CategoryRepository categoryRepository;
+
+  /**
+   * 요청된 카테고리 ID의 경로를 반환합니다.
+   *
+   * @param categoryId 요청된 카테고리 ID
+   * @return 카테고리 경로 문자열
+   */
+  public String buildPath(Long categoryId) {
+    if (categoryId == null) return null;
+    List<Flat> flats = categoryRepository.findAllFlat();
+    Map<Long, Flat> all = flats.stream()
+        .collect(Collectors.toMap(Flat::getId, f -> f));
+    return pathOf(categoryId, all, new HashMap<>());
+  }
+
+  /**
+   * 요청된 카테고리 ID들의 경로를 반환합니다.
+   *
+   * @param categoryIds 요청된 카테고리 ID들의 집합
+   * @return 카테고리 ID와 해당 경로의 맵
+   */
+  public Map<Long, String> buildPathsFor(Set<Long> categoryIds) {
+    if (categoryIds == null || categoryIds.isEmpty()) return Collections.emptyMap();
+
+    List<Flat> flats = categoryRepository.findAllFlat();
+    Map<Long, Flat> all = flats.stream()
+        .collect(Collectors.toMap(Flat::getId, f -> f));
+
+    Map<Long, String> memo = new HashMap<>();
+    for (Long id : categoryIds) pathOf(id, all, memo);
+
+    // 요청된 것만 반환
+    Map<Long, String> result = new HashMap<>();
+    for (Long id : categoryIds) result.put(id, memo.get(id));
```

Comment:
가능하면 아래처럼 할당과 선언을 동시에 하는 형태로 구현하시면 좋을 것 같아요. list를 만들고 put을 하는 형태는 지양되고 있습니다(가능하면 final화 하여 코드 작성)
```suggestion
    categoryIds.stream().collect(Collectors.toMap(id -> id, id -> memo.get(id)));
```